pkg/cnb/cnb_metadata.go

```diff
@@ -24,67 +26,63 @@ type FetchableBuilder interface {
 	ImagePullSecrets() []v1.LocalObjectReference
 }
 
+type ImageFetcher interface {
+	Fetch(keychain authn.Keychain, repoName string) (ggcrv1.Image, string, error)
+}
+
 type RemoteMetadataRetriever struct {
-	RemoteImageFactory registry.RemoteImageFactory
+	KeychainFactory registry.KeychainFactory
+	ImageFetcher    ImageFetcher
 }
 
 func (r *RemoteMetadataRetriever) GetBuilderImage(builder FetchableBuilder) (v1alpha1.BuilderRecord, error) {
-	img, err := r.RemoteImageFactory.NewRemote(builder.Image(), registry.SecretRef{
+	builderKeychain, err := r.KeychainFactory.KeychainForSecretRef(registry.SecretRef{
 		Namespace:        builder.GetObjectMeta().GetNamespace(),
 		ImagePullSecrets: builder.ImagePullSecrets(),
 	})
 	if err != nil {
-		return emptyRecord, errors.Wrap(err, "unable to fetch remote builder baseImage")
+		return v1alpha1.BuilderRecord{}, errors.Wrap(err, "unable to create builder image keychain")
```

Comment:
You don't like the `emptyRecord`? 😞

Reply:
I feel like the idiom I've seen is to create the empty struct right then and there. Creating a const totally works, but feel like more work I suppose.